marklogic-client-api/src/main/java/com/marklogic/client/impl/PlanBuilderSubImpl.java

```diff
@@ -560,13 +560,13 @@ public PlanParamExpr param(XsStringVal name) {
 
     return mapdef;
   }
-  static Map<String,String> makeMap(PlanSampleByOptions options) {
+  static Map<String,XsIntVal> makeMap(PlanSampleByOptions options) {
```

Comment:
I have concern about the ability to add other options in the future.  Options are typically string values.

If the limit is a string, will the server cast it to a number?  

If so, I'd suggest that String is a better choice for the data type of the map value by serializing the number to a string before assigning to the map.

If the server doesn't cast, maybe the right approach is to use int for now but to switch to string once the server can cast.

Reply:
Hi Erik, the server doesn't cast, and I get exceptions because of it, "Server Message: OPTIC-INVALARGS: fn.error(null, 'OPTIC-INVALARGS', 'limit must be a positive number: '+length); -- Invalid arguments: limit must be a positive number: 4"

I have more comments in https://github.com/marklogic/java-client-api/issues/1315.

Reply:
Thanks for clarifying.  I'd agree with the changes for now but think that it would be good to add a bug on the Optic SJS code to try to cast a string to a number if possible so the Java client can be refactored later if other options become necessary.